agent/pkg/crane/k8s/deployment_test.go

```diff
@@ -0,0 +1,270 @@
+//go:build unit
+// +build unit
+
+package k8s_test
+
+import (
+	"testing"
+
+	v1 "github.com/dyrector-io/dyrectorio/agent/pkg/api/v1"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/dyrector-io/dyrectorio/agent/pkg/crane/config"
+	"github.com/dyrector-io/dyrectorio/agent/pkg/crane/k8s"
+
+	internalconfig "github.com/dyrector-io/dyrectorio/agent/internal/config"
+)
+
+func TestResourceParsingCorrect(t *testing.T) {
+	resourceConfig := v1.ResourceConfig{
+		Limits: v1.Resources{
+			CPU:    "1m",
+			Memory: "2m",
+		},
+		Requests: v1.Resources{
+			CPU:    "3m",
+			Memory: "4m",
+		},
+	}
+
+	resources, err := k8s.GetResourceManagementForTest(resourceConfig, nil)
+
+	assert.Nil(t, err)
+	assert.Equal(t, resources.Limits.Cpu().String(), "1m")
+	assert.Equal(t, resources.Limits.Memory().String(), "2m")
+	assert.Equal(t, resources.Requests.Cpu().String(), "3m")
+	assert.Equal(t, resources.Requests.Memory().String(), "4m")
+}
+
+func TestResourceParsingLimitsCPUError(t *testing.T) {
+	resourceConfig := v1.ResourceConfig{
+		Limits: v1.Resources{
+			CPU:    "ERROR",
+			Memory: "2m",
+		},
+		Requests: v1.Resources{
+			CPU:    "3m",
+			Memory: "4m",
+		},
+	}
+
+	_, err := k8s.GetResourceManagementForTest(resourceConfig, nil)
+
+	assert.Error(t, err)

```

Comment:
This is general, better would to be to assert the error itself, but here are no error types nor constants defined as the error texts, this also means. That it would be nice to have an error type for resource errors.

Reply:
😊